spec/rubocop/cop/style/endless_range_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::EndlessRange, :config do
+  subject(:cop) { described_class.new(config) }
+
+  context '<= Ruby 2.5', :ruby25 do
+    it 'reports no offense for array slicing with -1' do
+      expect_no_offenses(<<~RUBY)
+        ary[1..-1]
+      RUBY
+    end
+  end
+
+  context '>= Ruby 2.6', :ruby26 do
+    it 'reports an offense for slicing to ..-1' do
+      expect_offense(<<~RUBY)
+        ary[1..-1]
+            ^^^^^ Prefer endless ranges (start..) for slicing till the end of array.
+      RUBY
+    end
+
+    it 'reports no offense for excluding end' do
+      expect_no_offenses(<<~RUBY)
+        ary[1...-1]
+      RUBY
+    end
+
+    it 'reports no offense for other methods' do
+      expect_no_offenses(<<~RUBY)
+        ary.push(1...-1)
+      RUBY
+    end
+
+    it 'reports no offense for array with range inside' do
+      expect_no_offenses(<<~RUBY)
+        ranges = [1...-1]
+      RUBY
+    end
+
+    it 'autocorrects' do
```

Comment:
You can handle auto-correction in the same method where you check for the offense by using `expect_correction`.
```ruby
expect_offense(<<~RUBY)
  ary[1..-1]
      ^^^^^ Prefer endless ranges (start..) for slicing till the end of array.
RUBY

expect_correction(<<~RUBY)
  ary[1..]
RUBY
```

Reply:
Oh, cool, thanks. Fixed.